chainer/functions/math/sparse_matmul.py

```diff
@@ -0,0 +1,512 @@
+import numpy
+
+import chainer
+from chainer.backends import cuda
+from chainer import function_node
+from chainer import utils
+from chainer.utils import type_check
+
+try:
+    from scipy import sparse
+    _scipy_available = True
+except ImportError:
+    _scipy_available = False
+
+
+class SparseCooMatrix(object):
+
+    def __init__(self, data, row, col, shape, use_variable=False):
+        self.data = data
+        if use_variable:
+            self.data = chainer.Variable(self.data)
+        self.row = row
+        self.col = col
+        self.shape = shape  # (row, col)
```

Comment:
Can we afford any data validation here, e.g. since `to_dense` depends on `ndim`.

Reply:
All right, I will add some validation code here for `shape` among others.